src/app/pokedex/pokedex-container.tsx

```diff
@@ -0,0 +1,68 @@
+"use client";
+
+import { Pokemon } from "@/models/pokemon";
+import { firebaseService } from "@/services/firebase-service";
+import PokedexTile from "./pokedex-tile";
+import { getPokemonIconUrl } from "./pokedex-helpers";
+import { useState, useEffect } from "react";
+
+export default function PokedexContainer() {
+    const [pokemon, setPokemon] = useState<Pokemon[]>([]);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<string | null>(null);
+
+    let pageContent: React.ReactElement[] = [<div key="loading" className="centered">Loading...</div>];
+
+    useEffect(() => {
+        const fetchData = async () => {
+            setLoading(true);
+            setError(null);
+
+            try {
+                await firebaseService.getPokemon().then((pokemonList) => {
+                    setPokemon(pokemonList);
+                    setLoading(false);
```

Comment:
you dont need the set loading here, having it in the finally is enough

Reply:
I've refactored this quite a bit. Now the state defaults to loading:

`const [loading, setLoading] = useState(true);`

and when useEffect runs, it will be set to false only if Pokemon were found without errors:

```
  useEffect(() => {
    // If initialPokemon is empty and no error, show loading
    if (!!pokemon.length && !error) {
      setLoading(false);
    }
  }, [pokemon, error]);
```

LMK what you think

Reply:
way cleaner,  i like it